### DISCUSSION TITLE
Hitting "Connection request declined"

### DISCUSSION THREAD

**Discussion by yenkhoon** [WalletConnect/walletconnect-monorepo#3078] [Developer Support]
We have implemented walletconnect using "@web3modal/react": "^2.0.0-rc.3",
while connecting to browser metamask we having no problem, however if we use the ios latest version of metamask to scan the qr code, we hit the error "Connection request declined"

Any suggestion on fixes or is there any deprecated function support?

**Comment by crypblizz8** [comment]
Hey there, it looks like a very old package. Our latest as of [writing](https://www.npmjs.com/package/@web3modal/react?activeTab=versions) is at `2.6.2`. Please upgrade and let us know how that works for you.

**Reply by yenkhoon** [reply]
upgraded, it required typescript version 5.1 +, after upgraded our react-scripts has to be override to version 5 typescript for work around. Do you have a proper guidelines on a proper solution for that?
